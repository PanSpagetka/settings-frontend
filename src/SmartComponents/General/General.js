import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';

import { Main, PageHeader, PageHeaderTitle } from '@redhat-cloud-services/frontend-components';

class General extends Component {

    render() {
        return (
            <React.Fragment>
                <PageHeader>
                    <PageHeaderTitle title='General'/>
                </PageHeader>
                <Main>
                    <h1> Sample Component </h1>
                </Main>
            </React.Fragment>
        );
    }
}

export default withRouter(General);
