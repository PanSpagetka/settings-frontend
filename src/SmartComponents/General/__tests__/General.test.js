import React from 'react';
import { mount } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import toJson from 'enzyme-to-json';
import General from '../General';

const mockStore = configureMockStore();
const store = mockStore({});

describe('General component', () => {
    it('should render correctly', () => {
        const wrapper = mount(
            <Provider store={ store }>
                <Router>
                    <General />
                </Router>
            </Provider>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
