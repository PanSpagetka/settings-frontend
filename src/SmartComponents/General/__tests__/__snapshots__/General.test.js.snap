// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`General component should render correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <withRouter(General)>
        <Route>
          <General
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "createHref": [Function],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            location={
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              }
            }
            match={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
          >
            <PageHeader>
              <section
                className="pf-l-page-header pf-c-page-header pf-l-page__main-section pf-c-page__main-section pf-m-light"
                widget-type="InsightsPageHeader"
              >
                <PageHeaderTitle
                  title="General"
                >
                  <Title
                    className=""
                    size="2xl"
                    widget-type="InsightsPageHeaderTitle"
                  >
                    <h1
                      className="pf-c-title pf-m-2xl"
                      widget-type="InsightsPageHeaderTitle"
                    >
                       
                      General
                       
                    </h1>
                  </Title>
                </PageHeaderTitle>
              </section>
            </PageHeader>
            <Connect(Main)>
              <Main>
                <section
                  className="pf-l-page__main-section pf-c-page__main-section"
                  page-type=""
                >
                  <h1>
                     Sample Component 
                  </h1>
                </section>
              </Main>
            </Connect(Main)>
          </General>
        </Route>
      </withRouter(General)>
    </Router>
  </BrowserRouter>
</Provider>
`;
